src/meta_type.rs

```diff
@@ -12,6 +12,11 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::sync::{
```

Comment:
Remember that this needs to work inside of Substrate runtimes which are `no_std`.

Reply:
A right. Then this won't fly. Not sure how we can pull this off then...

Reply:
Indeed it is a tricky problem, needs some thinking about...

Reply:
Maybe I can get this to work by implementing it as an n-ary function, I'm experiment RN